src/sources/kubernetes_logs/parser/cri.rs

```diff
@@ -27,86 +29,144 @@ pub const NEW_LINE_TAG: &str = "new_line_tag";
+        // Get the log field with the message, if it exists, and coerce it to bytes.
+        let log = event.as_mut_log();
+        let value = log.get(self.field).map(|s| s.coerce_to_bytes());
+        match value {
+            None => {
+                // The message field was missing, inexplicably.
+                //
+                // This is definitely wrong so make sure we track this. We'll still ultimately
+                // end up passing the event along, though.
+                emit!(ParserMissingFieldError { field: self.field });
+                return;
             }
+            Some(s) => match self.pattern.captures_read(&mut self.capture_locations, &s) {
+                None => {
+                    emit!(ParserMatchError { value: &s[..] });
+                    return;
+                }
+                Some(_) => {
+                    let locations = &self.capture_locations;
+                    let captures = self.capture_names.iter().filter_map(|(idx, name)| {
+                        locations.get(*idx).and_then(|(start, end)| {
+                            let raw = &s[start..end];
+
+                            // For all fields except `timestamp`, simply treat them as `Value::Bytes`. For
+                            // `timestamp`, however, we actually make sure we can convert it correctly and feed it
+                            // in as `Value::Timestamp`.
+                            let value = if name == TIMESTAMP_TAG {
+                                let ds = String::from_utf8_lossy(raw);
+                                match DateTime::parse_from_str(&ds, "%+") {
+                                    Ok(dt) => Some(Value::Timestamp(dt.with_timezone(&Utc))),
+                                    Err(e) => {
+                                        emit!(ParserConversionError {
+                                            name,
+                                            error: conversion::Error::TimestampParse {
+                                                s: ds.to_string(),
+                                                source: e,
+                                            },
+                                        });
+                                        None
+                                    }
+                                }
+                            } else {
+                                Some(Value::Bytes(Bytes::copy_from_slice(raw)))
+                            };
+
+                            value.map(|v| (name, v))
+                        })
+                    });
+
+                    let mut drop_original = true;
+                    for (name, value) in captures {
+                        // If we're already overriding the original field, don't remove it after.
+                        if name == self.field {
+                            drop_original = false;
+                        }
+
+                        let _ = log.insert(name.as_str(), value);
```

Comment:
Is the `let _ =` here is to avoid a "must use" warning? Would writing that as `drop(…)` be clearer?

Reply:
It's not, just habit to surface that we're potentially getting back something that we will immediately throw away. I can change it to `drop(...)`.

Reply:
Addressed in b1ec38102.

Reply:
The more I think about it, the less I think these markings have much value. Also, marking all such call sites is going to be a Sisyphean task without `#[must_use]`, which we of course can't inject into external crates. I don't mind much either way in this case, but I think it will be a reduction to readability to require such a marking everywhere. Am I off base?

Reply:
Yeah, dunno.

While normally the standard library would be the default best practice to follow, even they don't have `HashMap::insert` marked as `#[must_use]`, although part of me feels weird about that because it's not obvious that it replaces the old value if it exists... and we don't have any doc comments on these methods to indicate that as such.